### DISCUSSION TITLE
Performance .where() vs policies

### DISCUSSION THREAD

**Discussion by Sesa1988** [supabase/supabase#8497] [Questions] [answered]
Hi,

My simplified scenario is:

- I want to query one table based on ownership of data and have a policy that checks that (uid()=user_id).
- The next step extends this functionality that everyone who owns this data **OR** if its marked as "public" can query.

This means my policy would be something like uid() = user_id OR is_public = true.

This means I would always get both datasets and have to use .where() in specific scenarios.

**My question is:** Is there a performance impact and is there a better way to do it? I think if there is no big performance impact it would be fine.

**Comment by MJDeligan** [comment, marked as answer]
Hey there,

Policies run on every query so there's obviously _some_ very slight "performance loss",  but they are inevitable since they do not simply replace where-clauses and where-clauses do not replace policies. In your frontend you can restrict the access with where-clauses, but as soon as a user gets hold of your frontend they have the anon-key (assuming you are not using middleware from which you query your db exclusively) and can query and manipulate any data you have not restricted access to. 

If you need to restrict access to data on every query like in your case (I guess) then policies are the way to go. Where-clauses are of course still needed to then limit the results you get on the frontend.

___
Policies created in supabase are permissive btw, meaning they are joined by ORs so you could just create a second policy with `is_public = true`.

Hope that helps.

**Reply by Sesa1988** [reply]
Thanks you the explaination, that would be a proper solution because I want just to split up some data on different parts of the UI like shared and own data but they are always able to query if its belongs to them or just public marked my another user on its data. 

I didnt know that you can even create a second policy for a select statement on the same table. That makes it more clean, thx :)
